microprofile/opentracing-smallrye/src/main/java/org/wildfly/microprofile/opentracing/smallrye/TracingCDIExtension.java

```diff
@@ -42,7 +45,18 @@ public static void registerApplicationTracer(ClassLoader classLoader, Tracer tra
 
     public void registerTracerBean(@Observes AfterBeanDiscovery abd) {
         abd.addBean().addTransitiveTypeClosure(Tracer.class).produceWith(i -> {
-            return TRACERS.get(Thread.currentThread().getContextClassLoader());
+            ClassLoader loader;
+            if (!WildFlySecurityManager.isChecking()) {
+                loader = Thread.currentThread().getContextClassLoader();
+            } else {
+                loader = AccessController.doPrivileged(new PrivilegedAction<ClassLoader>() {
+                    @Override
+                    public ClassLoader run() {
+                        return Thread.currentThread().getContextClassLoader();
+                    }
+                });
+            }
+            return TRACERS.get(loader);
```

Comment:
This could just be `TRACERS.get(WildFlySecurityManager.getCurrentContextClassLoader())`

Reply:
yes that was the issue